apps/webapp/src/state/send.ts

```diff
@@ -96,26 +91,31 @@ export const createSendSlice = (): SliceCreator<SendSlice> => (set, get) => {
   };
 };
 
-const getExponent = (asset: TempAsset): number => {
-  const match = asset.denomUnits.find(u => u.denom === asset.display);
-  return match?.exponent ?? 0;
-};
+const getExponent = (asset: AssetBalance): number => asset.denom.exponent;
+
+const planWitnessBuildBroadcast = async ({ amount, recipient, asset, account }: SendSlice) => {
+  if (typeof account === 'undefined') throw new Error('no selected account');
+  if (!asset) throw new Error('no selected asset');
+
+  const source = await getIndexByAddress(account);
```

Comment:
I don't think it's necessary to make the request if, in the dialog, the index is selected along with the asset. We should already know the index. Let's change the SendSlice to this:


```typescript
interface Selection {
  address: string;
  accountIndex: number;
  asset: AssetBalance;
}

export interface SendSlice {
  selection: Selection | undefined;
  ...

```

This way, if the user selected something, we always have access to the things that come with the selection without needing further requests.

Reply:
we do not have a randomize value for AddressIndex! or do they not need it?

Reply:
If we need to pass a source address, I don't think we need randomizer. The randomizer is only relevant for ephemeral addresses. Please test though.

Reply:
works! i'll delete this